.changeset/fluffy-dolls-sleep.md

```diff
@@ -0,0 +1,24 @@
+---
+'astro': minor
+---
+
+Adds a new way to configure the `i18n.locales` array.
+
+Developers can now assign a custom path that can span multiple language codes:
+
+```js
+// astro.config.mjs
+export default defineConfig({
+  experimental: {
+      i18n: {
+          defaultLocale: "english",
+          locales: [
+            { path: "english", codes: ["en", "en-US"]}
+          ],
```

Comment:
```suggestion
          locales: [
            "de",
            { path: "english", codes: ["en", "en-US"]},
            "fr",
          ],
```
If this is actually true ( :smile: ), then it could be helpful to show the mix of the two formats together?

If not true, and all `locales` entries must be of the same type, I wouldn't bother to actually explain that here, but I would show a second language using the same format to imply that, and nudge people to using one configuration or the other.